compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -58,23 +58,42 @@ func protoComponentsFromScanComponents(c *nodes.Components) *scannerV1.Component
 	return protoComponents
 }
 
-func convertRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELComponent {
+func convertAndDedupRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELComponent {
 	if rc == nil || rc.Packages == nil {
 		log.Warn("No RHEL packages found in scan result")
 		return nil
 	}
-	v1rhelc := make([]*scannerV1.RHELComponent, 0, len(rc.Packages))
-	for _, rhelc := range rc.Packages {
-		v1rhelc = append(v1rhelc, &scannerV1.RHELComponent{
-			// TODO(ROX-13936): Find out if ID field is needed here
+
+	convertedComponents := make(map[string]*scannerV1.RHELComponent, 0)
+	for i, rhelc := range rc.Packages {
+		comp := &scannerV1.RHELComponent{
+			// The loop index is used as ID, as this field only needs to be unique for each NodeInventory result slice
+			Id:          int64(i),
 			Name:        rhelc.Name,
 			Namespace:   rc.Dist,
 			Version:     rhelc.Version,
 			Arch:        rhelc.Arch,
 			Module:      rhelc.Module,
 			Cpes:        rc.CPEs,
 			Executables: rhelc.Executables,
-		})
+		}
+		compKey := makeComponentKey(comp)
+		if compKey != "" {
+			if _, contains := convertedComponents[compKey]; !contains {
+				log.Debugf("Adding component %v to convertedComponents", comp.Name)
+				convertedComponents[compKey] = comp
+			} else {
+				log.Warnf("Detected package collision in Node Inventory scan. Skipping package %v at index %v", comp, i)
```

Comment:
Why not being more specific than `%v`?
```suggestion
				log.Warnf("Detected package collision in Node Inventory scan. Skipping package %s at index %d", compKey, i)
```